protocol/messenger.go

```diff
@@ -204,6 +204,30 @@ type EnvelopeEventsInterceptor struct {
 	Messenger             *Messenger
 }
 
+func (m *Messenger) CanonicalName() (string, error) {
```

Comment:
@mohsen-ghafouri, I'm confused by the names of the functions. One is `CanonicalName` and the other is `GetCanonicalName`, both get the same "thing" based on their name, but one returns either the ENS name or display name and the other the primary name.

So in the end, what is a canonical name? Should the `m.settings.GetPreferredUsername` be part of the logic to return a canonical name?

Reply:
actually `CanonicalName` is for current user and should return user name/ens/nickname, and i gave priority to ens if it has any. `GetCanonicalName` func but return contact primaryName by mentionId. i can change `CanonicalName` to for example `GetUserPrefferredName` maybe? @ilmotta

Reply:
I pushed some changes please check again. thanks

Reply:
Thanks @mohsen-ghafouri, but I still don't understand from the code alone why the canonical name of the current account means something different than the canonical name of some other contact.

The function `GetCanonicalName` could be read as `GetCanonicalNameIfCurrentAccountOtherwisePrimaryName`. Of course this is an absurd function name, but I'm just trying to illustrate why it's so confusing to me. The point is that `GetCanonicalName` doesn't return a canonical name if it's not the current user, it returns the PrimaryName, two very different things, so why name both as if they are the same thing?

I hope my confusion makes some sense :)

Reply:
if it's all about func name, what do you think about `GetOwnCanonicalName` and `ResolveCanonicalName` for the second func ?

Reply:
@mohsen-ghafouri, even though I'm not new to the code, the naming and what they mean is still confusing to me, like primary name, secondary name, display name, preferred name, nickname, it's a lot of *names, and now we'll have "canonical name" :) ENS name is clear of course. Maybe others in the team are also confused just as me...

Maybe @cammellos can help here?

Reply:
We can rename to `PrimaryName` -> intead of `Canonical`?
The only difference between the user logged in and any contacts, is that in contacts you might have a `LocalNickname`, which takes precendence over ENS/Display name, while for that's not available for yourself (i.e you can nickname yourself), and therefore it's ENS/Display name

Would that clear the confusion?

Reply:
thanks, so i can rename func to `GetOwnPrimaryName` and `ResolvePrimaryName` , i hope that it can help

Reply:
It cleared up a little bit, thanks. I found this unit test which made me better understand your suggestion @cammellos https://github.com/status-im/status-go/blob/7d03ae8272f856cb4a7ac6539a3d295b10b3b269/protocol/contact_test.go#L727

Reply:
thanks for your suggestion, resolved